jax/experimental/export/export.py

```diff
@@ -826,14 +827,63 @@ def walk_operations(op):
            "See https://github.com/google/jax/blob/main/jax/experimental/jax2tf/README.md#native-lowering-supports-only-select-custom-calls")
     raise ValueError(msg)
 
+def expand_in_shardings(in_shardings: tuple[Sharding, ...],
+                        module_kept_var_idx: Sequence[int],
+                        nr_inputs: int) -> tuple[Sharding, ...]:
+  """Expands in_shardings with unspecified shardings for inputs not kept.
+
+  Assumes in_shardings corresponds to module_kept_var_idx.
+  """
+  assert len(in_shardings) == len(module_kept_var_idx)
+  assert nr_inputs >= len(module_kept_var_idx)
+  all_in_shardings: list[Sharding] = [sharding_impls.UNSPECIFIED] * nr_inputs
+  for idx, in_s in zip(sorted(module_kept_var_idx), in_shardings):
+    all_in_shardings[idx] = in_s
+  return tuple(all_in_shardings)
+
+
+def canonical_shardings(
```

Comment:
Maybe add a TODO somewhere with your and/or my name saying "Remove the conversion once jit supports UNSPECIFIED along with concrete shardings to {in|out}_shardings"

Reply:
Done